### PR TITLE
fix(upgrade): propagate --all and queue-scoping flags to auto init --upgrade retry

### PR DESCRIPTION
**Pull request by rios0rios0**
## Summary

- Fixes a bug where the reactive `init --upgrade` retry inside `UpgradeAwareShellRepository` dropped `--all` (and the related `--filter` / `--queue-*` flags), so a queued `terragrunt apply --all` run that failed with `Module not installed` retried as bare `terragrunt init --upgrade` in the parent directory and aborted with `You attempted to run terragrunt in a folder that does not contain a terragrunt.hcl file`.
- The retry now reproduces the original queue scoping so init walks the same units and the original command is retried successfully.
- Forwarded flags (both space and `--flag=value` forms): `--all`, `--filter`, `--queue-include-dir`, `--queue-exclude-dir`, `--queue-include-units-reading`, `--queue-strict-include`, `--queue-include-external`, `--queue-exclude-external`.

## Failure pattern this fixes

A queued run such as `terragrunt apply --all <queue-scoping flags> --non-interactive -auto-approve` from a parent directory that has no `terragrunt.hcl` would:

1. Fail with `Module not installed` because units inside the queue had not been initialized.
2. Trigger terra's auto-init detector, which then ran a bare `terragrunt init --upgrade` in that same parent directory.
3. Abort with `You attempted to run terragrunt in a folder that does not contain a terragrunt.hcl file. Please add a terragrunt.hcl file and try again.`

After this PR the retry preserves the queue scoping (e.g. `terragrunt init --upgrade --all <queue-scoping flags>`) so terragrunt walks the same set of units instead of looking for a `terragrunt.hcl` in the parent directory.

## Implementation notes

- New `extractQueueScopingFlags` helper in `internal/infrastructure/repositories/upgrade_shell_repository.go` uses a Mapper-style map keyed by spec name (no switch/case), preserves the original relative order, and handles end-of-list valued flags safely.
- `runInitUpgrade` now takes the original arguments and prepends the extracted queue flags to `init --upgrade`. Single call site in `ExecuteCommandWithUpgrade` updated.
- Proactive init (`shouldInitEnvironment`) was already correct — it skips when `--all` is present, so no change there.

## Test plan

- [x] `go test -tags unit ./...` — all packages pass
- [x] BDD-style cases for `extractQueueScopingFlags` (no flags, bare `--all`, space-form valued flag, `--flag=value` form, `--filter` in both forms, mixed flags with order preservation, end-of-list valued flag)
- [x] End-to-end retry test that fails the first `apply --all <queue flags>` with the upgrade pattern, then asserts the wrapped script receives both flags on the `init --upgrade` retry
- [x] `make lint` introduces no new findings

## :vertical_traffic_light: Quality checklist

- [x] Did you add the changes in the `CHANGELOG.md`?
- [x] Did you run all the code checks? (`go test`)
- [x] Are the tests passing?

### CODE REVIEW

File: internal/infrastructure/repositories/upgrade_shell_repository.go [hunk 1]
```diff
@@ -182,6 +188,70 @@ func (it *UpgradeAwareShellRepository) runInitUpgrade(
...
+			index++
+		}
+	}
+	return forwarded
+}
+
+// ExtractQueueScopingFlagsPublic is a public wrapper for testing the private
+// extractQueueScopingFlags function.
+func ExtractQueueScopingFlagsPublic(arguments []string) []string {
+	return extractQueueScopingFlags(arguments)
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/infrastructure/repositories/upgrade_shell_repository.go]: Addressed in `375d6e4`. Removed the `ExtractQueueScopingFlagsPublic` wrapper and replaced the 7 helper-level unit tests with 4 behavior-level tests that exercise the same edge cases (`--queue-exclude-dir=value` form, `--filter <query>` space form, and a negative case proving unrelated flags like `-auto-approve` are not forwarded to `init --upgrade`) through the public `ExecuteCommandWithUpgrade` API. Production package surface stays unchanged.
